infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerMachine.java

```diff
@@ -148,6 +150,14 @@ void setServerStatus(String serverRef, ServerStatus status) {
         server.setStatus(status);
     }
 
+    public void putResource(String targetPath, InputStream sourceStream) throws InfrastructureException {
+        try {
+            docker.putResource(PutResourceParams.create(container, targetPath, sourceStream));
+        } catch (IOException e) {
+            throw new InfrastructureException(e.getMessage(), e);
```

Comment:
It is supposed that exception that is probable and expected and say nothing helpful to admin is InfrastructureException, whereas InternalInfrastructureException should take attention of admin as unusual thing that may identify some problem in product